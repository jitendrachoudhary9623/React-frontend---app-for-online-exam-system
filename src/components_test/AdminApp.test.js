import React, { Component } from 'react';
import AdminApp from "../components/admin/AdminApp.component"
import { shallow,mount } from 'enzyme'
import { Provider } from 'react-redux';
import { store } from '../datasource/dataStore';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store';

describe('AdminApp', () => {
    it('should be defined', () => {
      expect(AdminApp).toBeDefined();
    });
    it('should render correctly', () => {
      const tree = shallow(
        <AdminApp/>
      );
      expect(tree).toMatchSnapshot();
    });

    it('Admin App should have SideBar', () => {
      const wrapper = shallow(
        <AdminApp/>
      );
      expect(wrapper.find('SideBar')).toBeDefined();
    });

    it('Admin App should have NavBar', () => {
      const wrapper = shallow(
        <AdminApp/>
      );
      expect(wrapper.find('NavBar')).toBeDefined();
    });

    it('Admin App card with p6 should have welcome admin text', () => {
      const wrapper = shallow(
        <AdminApp/>
      );
      const cardText="Welcome Admin";
      expect(wrapper.find('.p6').text()).toEqual(cardText);
    });

    it('Admin App should have container class', () => {
      const wrapper = shallow(
        <AdminApp/>
      );
      expect(wrapper.find('.container')).toBeDefined();
    });
   });