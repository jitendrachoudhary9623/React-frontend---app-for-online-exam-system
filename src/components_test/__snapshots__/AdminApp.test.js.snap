exports[`AdminApp should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminApp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          isLoggedIn={true} />,
        <div
          className="row">
          <div
                    className="col s2"
                    style={
                              Object {
                                        "backgroundColor": "white",
                                        "height": "87vh",
                                      }
                    }>
                    <Unknown />
                              
          </div>
          <div
                    className="col s10">
                    <div
                              className="container">
                              <div
                                        className="card p6">
                                        Welcome Admin
                              </div>
                    </div>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isLoggedIn": true,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(NavBar)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col s2"
              style={
                            Object {
                                          "backgroundColor": "white",
                                          "height": "87vh",
                                        }
              }>
              <Unknown />
                        
</div>,
            <div
              className="col s10">
              <div
                            className="container">
                            <div
                                          className="card p6">
                                          Welcome Admin
                            </div>
              </div>
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown />,
                "          ",
              ],
              "className": "col s2",
              "style": Object {
                "backgroundColor": "white",
                "height": "87vh",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(SideBar)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              "          ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container">
                <div
                                className="card p6">
                                Welcome Admin
                </div>
</div>,
              "className": "col s10",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card p6">
                  Welcome Admin
</div>,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome Admin",
                  "className": "card p6",
                },
                "ref": null,
                "rendered": "Welcome Admin",
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isLoggedIn={true} />,
          <div
            className="row">
            <div
                        className="col s2"
                        style={
                                    Object {
                                                "backgroundColor": "white",
                                                "height": "87vh",
                                              }
                        }>
                        <Unknown />
                                  
            </div>
            <div
                        className="col s10">
                        <div
                                    className="container">
                                    <div
                                                className="card p6">
                                                Welcome Admin
                                    </div>
                        </div>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isLoggedIn": true,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(NavBar)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col s2"
                style={
                                Object {
                                                "backgroundColor": "white",
                                                "height": "87vh",
                                              }
                }>
                <Unknown />
                          
</div>,
              <div
                className="col s10">
                <div
                                className="container">
                                <div
                                                className="card p6">
                                                Welcome Admin
                                </div>
                </div>
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown />,
                  "          ",
                ],
                "className": "col s2",
                "style": Object {
                  "backgroundColor": "white",
                  "height": "87vh",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(SideBar)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  },
                },
                "          ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="container">
                  <div
                                    className="card p6">
                                    Welcome Admin
                  </div>
</div>,
                "className": "col s10",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="card p6">
                    Welcome Admin
</div>,
                  "className": "container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Welcome Admin",
                    "className": "card p6",
                  },
                  "ref": null,
                  "rendered": "Welcome Admin",
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
